zerver/tests/test_markdown.py

```diff
@@ -2950,27 +2962,6 @@ def assert_silent_mention(content: str) -> None:
         assert_silent_mention("```quote\n@*backend*\n```")
         assert_silent_mention("```quote\n@_*backend*\n```")
 
-    def test_system_user_group_mention(self) -> None:
-        desdemona = self.example_user("desdemona")
-        iago = self.example_user("iago")
-        hamlet = self.example_user("hamlet")
-        moderators_group = NamedUserGroup.objects.get(
-            realm=iago.realm, name=SystemGroups.MODERATORS, is_system_group=True
-        )
-        content = "@*role:moderators* @**King Hamlet** test message"
-
-        # Owner cannot mention a system user group.
-        msg = Message(sender=desdemona, sending_client=get_client("test"), realm=desdemona.realm)
-        rendering_result = render_message_markdown(msg, content)
-        self.assertEqual(rendering_result.mentions_user_ids, {hamlet.id})
-        self.assertNotIn(moderators_group.id, rendering_result.mentions_user_group_ids)
-
-        # Admin belonging to user group also cannot mention a system user group.
-        msg = Message(sender=iago, sending_client=get_client("test"), realm=iago.realm)
-        rendering_result = render_message_markdown(msg, content)
-        self.assertEqual(rendering_result.mentions_user_ids, {hamlet.id})
-        self.assertNotIn(moderators_group.id, rendering_result.mentions_user_group_ids)
```

Comment:
Removed this test since we not allow mentioning system groups but that is being controlled by `can_mention_group` setting and `render_message_markdown` does not check the setting, it is checked in message sending codepath.